File: Queen.java

import java.util.ArrayList;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Pablo
 */
public class Queen extends Piece {
    // constructor
	public Queen( int color){
		m_color = color;
		
		if (color==0) m_type = Utils.wQueen;
		else m_type = Utils.bQueen;
		
	}
	
	
	// this method must be completed with all the possible pieces
	
	public ArrayList<Action> getPossibleActions(State state){
		ArrayList<Action> list = null;
		
		list = this.getHorizontalLeftMoves(state);
		list.addAll(this.getHorizontalRightMoves(state));
		list.addAll(this.getVerticalDownMoves(state));
		list.addAll(this.getVerticalUpMoves(state));
                list.addAll(this.getDiagonalLeftDownMoves(state));
                list.addAll(this.getDiagonalRightDownMoves(state));
		list.addAll(this.getDiagonalLeftUpMoves(state));
		list.addAll(this.getDiagonalRightUpMoves(state));
		
		return list;
	}
}

File: Queen.py
import Utils
from Piece import Piece

# this class implements the getPossibleActions for each type of piece

class Queen(Piece): 
	
	# constructor
	def __init__(self, color):
		self.m_color = color
		
		if color==0: 
			self.m_type = Utils.wQueen
		else:
			self.m_type = Utils.bQueen

	
	# this method must be completed with all the possible pieces	
	def getPossibleActions(self, state):
		l = []
		
		l = self.getDiagonalLefUptMoves(state)
		l += self.getDiagonalRightDownMoves(state)
		l += self.getDiagonalLeftDownMoves(state)
		l += self.getDiagonalRightUpMoves(state)
        l += self.getHorizontalLeftMoves(state)
		l += self.getHorizontalRightMoves(state)
		l += self.getVerticalDownMoves(state)
		l += self.getVerticalUpMoves(state)
		
		return l